src/terrain/KM_Terrain.pas

```diff
@@ -215,6 +242,8 @@   TKMTerrain = class
     procedure UpdatePassability(aRect: TKMRect); overload;
     procedure UpdatePassability(Loc: TKMPoint); overload;
 
+    procedure UpdateWalkConnect(const aSet: array of TWalkConnect; aRect: TKMRect; aDiagObjectsEffected: Boolean);

```

Comment:
Why did you make this public? It's an internal terrain thing so lets leave it private.